Support sphere in SetSample

sf1919 opened the pull request:
**Description of work.**
Previously anyone wanting to use a sphere in  SetSample would have to do so using an xml string. This PR adds sphere to the preset shapes available.

**To test:**

1. Build Mantid
2. Run the following code and it shouldn't crash
```
ws = CreateSampleWorkspace()
SetSample(ws,
        Geometry={'Shape': 'Sphere',
                  'Radius': 2.0, 'Center': [0.,0.,0.]})
```
3. Code review

Fixes #32892 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Added Sphere to preset shapes in SetSample

sf1919 committed:
Added test for Sphere preset shape

sf1919 committed:
Added documentation

sf1919 committed:
Fixed cpp error

sf1919 committed:
Removed unused reference

DannyHindson commented on Framework/DataHandling/src/SetSample.cpp:
```diff
@@ -31,6 +31,7 @@
 #include <Poco/Path.h>
 #include <boost/algorithm/string/case_conv.hpp>
 #include <boost/algorithm/string/predicate.hpp>
+#include <boost/range/algorithm/transform.hpp>
```
I don't think you need this include. Might be a feature from the original PR for this change but compiles for me without it

DannyHindson commented on docs/source/algorithms/SetSample-v1.rst:
```diff
@@ -45,6 +45,7 @@ be in centimeters):
 - ``HollowCylinder``: Height, InnerRadius, OuterRadius, Center
 - ``FlatPlateHolder``: Width, Height, Thick, Center, Angle, FrontThick, BackThick. This is a CSG union of 2 FlatPlates tightly wrapping a FlatPlate sample. To be used for the ContainerGeometry.
 - ``HollowCylinderHolder``: Height, InnerRadius, InnerOuterRadius, OuterInnerRadius, OuterRadius, Center. This is a CSG union of 2 HollowCylinders tightly wrapping a HollowCylinder sample. To be used for the ContainerGeometry.
+- ``Sphere``: Center, Radius
```
This is a bit lazy of me but I've been meaning to open an issue for another v minor doc change on L44 where the "Axis" property isn't mentioned in here despite being supported in the code (see SetSample::createCylinderLikeXML). Would you mind updating L44 and L45 to add this parameter in eg  `Cylinder: Height, Radius, Center, Axis`?
This came up on a call with the WISH team recently

DannyHindson commented on docs/source/algorithms/SetSample-v1.rst:
```diff
@@ -190,22 +191,16 @@ used for the dictionary parameters.
+
+   # Set sample geometry of workspace to a Sphere
+   SetSample(ws, Geometry={'Shape': 'Sphere',
+                 'Radius': 2.0, 'Center': [0.,0.,0.]})
```
Great! That's nice and simple now

DannyHindson reviewed (changes requested):
Just a few v minor things. Scripts works well and changes look good.

I extended your test script a bit to plot the shape using this and all looks fine:
```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np
from mpl_toolkits.mplot3d.art3d import Poly3DCollection

ws = CreateSampleWorkspace()
SetSample(ws,
        Geometry={'Shape': 'Sphere',
                  'Radius': 2.0, 'Center': [0.,0.,0.]})
                  
sample = ws.sample()
shape = sample.getShape()
mesh = shape.getMesh()

mesh_polygon = Poly3DCollection(mesh, facecolors = ['g'], edgecolors = ['b'], alpha = 0.5, linewidths=0.1)

fig, axes = plt.subplots(subplot_kw={'projection':'mantid3d', 'proj_type':'ortho'})
#fig, axes = plt.subplots(subplot_kw={'projection':'mantid3d'})
axes.add_collection3d(mesh_polygon)

axes.set_mesh_axes_equal(mesh)
axes.set_xlabel('X / m')
axes.set_ylabel('Y / m')
axes.set_zlabel('Z / m')

plt.show()
```

sf1919 committed:
Changes requested in review

- removed obsolete include statement
- updated documentation for using the preset Cylinder and HollowCylinder shapes

DannyHindson reviewed (approved):
All looks good now. Thanks for adding in the Axis doc change
